pkg/dmaasbackup/schedule.go

```diff
@@ -77,12 +83,159 @@ func (d *dmaasBackup) generateBackupName(dbkp v1alpha1.DMaaSBackup) string {
 // updateScheduleInfo checks for relevant velero schedule/backup and
 // update the dbkp with schedule/backup details
 func (d *dmaasBackup) updateScheduleInfo(dbkp *v1alpha1.DMaaSBackup) error {
-	// TODO
+	// check if dbkp is having config for non-scheduled backup
+	if yes := isSchedule(*dbkp); !yes {
+		bkp, err := d.backupClient.Get(dbkp.Name, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to update scheduleInfo")
+		}
+		if bkp.Name == "" {
+			// Backup could have deleted due to expiration of TTL
+			// TODO: Should we clear backupname from dmaasbackup? and/or recreate?
+			return nil
+		}
+		if dbkp.Status.VeleroBackupName == nil {
+			backupName := bkp.Name
+			dbkp.Status.VeleroBackupName = &backupName
```

Comment:
As user is not supposed to touch 'Status', can we change this to string? or is this pointer to reduce space on etcd?

Reply:
This is mainly to avoid string comparison to check if backup is set or not.

Reply:
ok